python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -2157,3 +2171,97 @@ def inject_env_vars(self, location_name: Optional[str]):
         new_env = self._secrets_loader.get_secrets_for_environment(location_name)
         for k, v in new_env.items():
             os.environ[k] = v
+
+    def get_logical_version_from_inputs(
```

Comment:
Because this contains business logic that's on top of the instance, I think it would make sense to put it closer to where it's used than as a method on `DagsterInstance`.

Reply:
The method used to be called from both core and dagster-graphql, but now I've removed it entirely